lib/Driver/Compilation.cpp

```diff
@@ -1196,6 +1221,17 @@ namespace driver {
         }
       };
 
+      // Load our priors, which are always adjacent to the build record. We
+      // don't care if this load succeeds or not. If it fails, and we succeed at
+      // integrating one of the external files below, the changeset will be the
+      // entire module!
+      const auto *externalPriorJob = Comp.addExternalJob(
+          std::make_unique<Job>(Comp.getDerivedOutputFileMap(),
+                                Comp.CompilationRecordPath + ".external"));
+      getFineGrainedDepGraph(forRanges).loadFromPath(
+          externalPriorJob, Comp.CompilationRecordPath + ".external",
+          Comp.getDiags());
+
```

Comment:
The comment is for line 1245: "Is this module out of date?"
Is that check just an optimization? If so could there be a unit test, or an assertion that says that without that date check, we still don't find any changes in the deps?

Reply:
Also, would it be better to move the `getEnableCrossModuleIncrementalBuild` check out of the for-loop?

Reply:
Come to think of it, the code might read better to factor out the body of the for-loop into a separate function. The 'continues' would become 'returns'. The function enclosing the for-loop might read better.

Reply:
> Is that check just an optimization?

That check mirrors the one we do for plain external dependencies.

> Also, would it be better to move the getEnableCrossModuleIncrementalBuild check out of the for-loop?

Actually, no. Think about what would happen if the user were to run their prior build with the flag enabled, then run this build with the flag disabled. They'll now have a set of incremental external dependencies that need to be treated like external dependencies, and to do that we'd need to extract the entire top half of this loop.

Reply:
Great point! Could it be in a comment? Otherwise some future person might try it and create a bug.